winforms/src/toga_winforms/app.py

```diff
@@ -61,33 +97,13 @@ def create(self):
         self.app_context = WinForms.ApplicationContext()
         self.app_dispatcher = Dispatcher.CurrentDispatcher
 
-        # Check the version of windows and make sure we are setting the DPI mode
-        # with the most up to date API
-        # Windows Versioning Check Sources : https://www.lifewire.com/windows-version-numbers-2625171
-        # and https://docs.microsoft.com/en-us/windows/release-information/
-        win_version = Environment.OSVersion.Version
-        if win_version.Major >= 6:  # Checks for Windows Vista or later
-            # Represents Windows 8.1 up to Windows 10 before Build 1703 which should use
-            # SetProcessDpiAwareness(True)
-            if (win_version.Major == 6 and win_version.Minor == 3) or (
-                win_version.Major == 10 and win_version.Build < 15063
-            ):
-                windll.shcore.SetProcessDpiAwareness(True)
-                print(
-                    "WARNING: Your Windows version doesn't support DPI-independent rendering.  "
-                    "We recommend you upgrade to at least Windows 10 Build 1703."
-                )
-            # Represents Windows 10 Build 1703 and beyond which should use
-            # SetProcessDpiAwarenessContext(-2)
-            elif win_version.Major == 10 and win_version.Build >= 15063:
-                windll.user32.SetProcessDpiAwarenessContext(-2)
-            # Any other version of windows should use SetProcessDPIAware()
-            else:
-                windll.user32.SetProcessDPIAware()
-
+        # These are required for properly setting up DPI mode
         self.native.EnableVisualStyles()
         self.native.SetCompatibleTextRenderingDefault(False)
```

Comment:
> However, the fonts' don't seem to be using the latest DPI value and as such they are not being scaled when a new DPI change is detected while the app is running.

That might be caused by these methods being called too late. [This page](https://learn.microsoft.com/en-us/dotnet/desktop/winforms/high-dpi-support-in-windows-forms?view=netframeworkdesktop-4.8#configuring-your-windows-forms-app-for-high-dpi-support) suggests calling `EnableVisualStyles` as "the first method call in your application entry point", before the main loop even starts. So they should probably be called at the same time as `SetProcessDpiAwarenessContext`.

Reply:
Yes but we need the Application handle to call EnableVisualStyles and the SetProcessDpiAwarenessContext call doesn't work when inside __init__() or create() methods.

I think we need some method like a refresh() for the fonts which can be called to redraw according to the new DPI Scale.

Reply:
`EnableVisualStyles` and `SetCompatibleTextRenderingDefault` are both static methods, so you can call them at any time.  It might not make any difference in this case, but it's probably a good idea to follow the documentation and call these methods before `SetProcessDpiAwarenessContext`, to avoid this question coming up again in the future.